dtran/pipeline.py

```diff
@@ -177,3 +257,52 @@ def save(self, save_path: Union[str, Path]):
     @staticmethod
     def load(load_path: Union[str, Path]):
         pass
+
+
+def default_wrapper(cls: Type[IFunc], inputs: Set[str]) -> Type[IFunc]:
+    class DefaultWrapper(IFunc):
+        func_cls = cls
+
+        def __init__(self, **kwargs):
+            # checking if the kwargs satisfy wrapped adapter's init function definition
+            try:
+                signature(DefaultWrapper.func_cls.__init__).bind(DefaultWrapper.func_cls, **kwargs)
+            except TypeError:
+                print(f"Cannot initialize cls: {DefaultWrapper.func_cls}")
+                raise
+            self.func_args = kwargs
+
+        async def exec(self) -> Union[dict, Generator[dict, None, None], AsyncGenerator[dict, None]]:
+            while True:
+                func_args = self.func_args.copy()
+                # waiting for all wired inputs, and break out of the loop if any one of them has finished
+                try:
+                    for argname in inputs:
+                        func_args[argname] = await self.func_args[argname].__anext__()
+                except StopAsyncIteration:
+                    break
+                func = self.func_cls(**func_args)
```

Comment:
why don't we move the part that create the instance of `func` to init? this would make implementing validate function easier

Reply:
I just fixed it. Must have been a typo.